File: Data Structures & Algorithms/house-robber/submission-11.java
class Solution {
    HashMap<Integer, Integer> map = new HashMap<>();
    public int rob(int[] nums) {
        int[] dp = new int[nums.length + 2];
        int unu = 0, doi = 0;
        for (int i = nums.length - 1; i>= 0; --i) {
            int temp = Math.max(nums[i] + doi, unu);
            doi = unu;
            unu = temp;
        }
        return unu;
    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-11.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int maxi = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] < 0) nums[i] = 0;
            maxi = std::max(maxi, nums[i]);
        }
        vector<int> v(maxi + 1, 0);
        for (int x : nums) {
            if (!v[x]) { 
                if (x == 0) {
                    v[x] = -nums.size();
                }
                else {
                    v[x] = -x;
                }
            }
        }
        for (int i = 1; i < v.size(); ++i) {
            if (v[i] == 0) return i; 
        }
        return v.size();
    }
};